scripts/qa_crowbarsetup.sh

```diff
@@ -2872,18 +2872,26 @@ function oncontroller_testsetup()
     tempestret=0
     if [ "$wanttempest" = "1" ]; then
         # Upload a Heat-enabled image
-        openstack image list | grep -q SLE11SP3-x86_64-cfntools || openstack image create \
-            --public --disk-format qcow2 --container-format bare --property hypervisor_type=kvm \
-            --copy-from http://$clouddata/images/SLES11-SP3-x86_64-cfntools.qcow2 \
-            SLE11SP3-x86_64-cfntools | tee glance.out
+        openstack image list | grep -q SLE11SP3-x86_64-cfntools || {
+            curl -s \
+                http://$clouddata/images/SLES11-SP3-x86_64-cfntools.qcow2 | \
+                openstack image create \
+                    --public --disk-format qcow2 --container-format bare \
+                    --property hypervisor_type=kvm \
+                    SLE11SP3-x86_64-cfntools
+        } | tee glance.out
         imageid=`perl -ne "m/ id [ |]*([0-9a-f-]+)/ && print \\$1" glance.out`
         crudini --set /etc/tempest/tempest.conf orchestration image_ref $imageid
         # test if is cnftools image prepared for tempest
-        wait_for 300 5 'glance image-show $imageid | grep active &>/dev/null' "prepare cnftools image"
+        wait_for 300 5 \
+            'openstack image show $imageid | grep active &>/dev/null' \
+            "prepare cnftools image"
         pushd /var/lib/openstack-tempest-test
         echo 1 > /proc/sys/kernel/sysrq
+        [ -x /usr/bin/tempest-cleanup ] && /usr/bin/tempest-cleanup --init-saved-state || :
         ./run_tempest.sh -N $tempestoptions 2>&1 | tee tempest.log
         tempestret=${PIPESTATUS[0]}
+        [ -x /usr/bin/tempest-cleanup ] && /usr/bin/tempest-cleanup --delete-tempest-conf-objects || :
         testr last --subunit | subunit-1to2 > tempest.subunit.log
         /var/lib/openstack-tempest-test/bin/tempest_cleanup.sh || :
```

Comment:
@vuntz yes, on todo (somewhere). removing it right now is sort of difficult as tempest-cleanup doesn't exist for cloud4.